test/boost/cql_query_test.cc

```diff
@@ -153,6 +153,74 @@ SEASTAR_TEST_CASE(test_twcs_max_window) {
     });
 }
 
+SEASTAR_TEST_CASE(test_twcs_restrictions_mixed) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        // Cenario 1: STCS->TWCS with no TTL defined
+        e.execute_cql("CREATE TABLE tbl (a int PRIMARY KEY, b int);").get();
+        e.execute_cql("ALTER TABLE tbl WITH compaction = {'class': 'TimeWindowCompactionStrategy'};").get();
+        e.require_table_exists("ks", "tbl").get();
+
+        // Cenario 2: STCS->TWCS with small TTL
+        e.execute_cql("CREATE TABLE tbl2 (a int PRIMARY KEY, b int) WITH default_time_to_live = 60;").get();
+        e.execute_cql("ALTER TABLE tbl2 WITH compaction = {'class': 'TimeWindowCompactionStrategy'};").get();
+        e.require_table_exists("ks", "tbl2").get();
+
+        // Cenario 3: STCS->TWCS with large TTL value
+        e.execute_cql("CREATE TABLE tbl3 (a int PRIMARY KEY, b int) WITH default_time_to_live = 8640000;").get();
+        e.require_table_exists("ks", "tbl3").get();
+        assert_that_failed(e.execute_cql("ALTER TABLE tbl3 WITH compaction = {'class': 'TimeWindowCompactionStrategy'};"));
+
+        // Cenario 4: TWCS table with small to large TTL
+        e.execute_cql("CREATE TABLE tbl4 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'} AND default_time_to_live = 60;").get();
+        e.require_table_exists("ks", "tbl4").get();
+        assert_that_failed(e.execute_cql("ALTER TABLE tbl4 WITH default_time_to_live = 86400000;"));
```

Comment:
In the code patches, I was wondering if we are handling the TWCS options coming from the pre-existing table correctly. Here I think this isn't checked properly - because you have a TWCS table with the default window option, so since it's the default you can't know if the empty options or the correct options are being checked.
I think you should explicitly list a TWCS window size in some of these checks (or add cases with an explict TWCS window size)

Reply:
By TWCS options I assume you mean the introduced restrictions? If so, yes, I agree to "hardcode" it on the test, specially should we want to change the defaults as you had mentioned a few comments earlier.

However, since your comment says  `from the pre-existing table correctly`, this got me confused. The code is quite simple: If the user is switching to TWCS and/or modifying `default_time_to_live` then we run the conditions. If the user is switching to TWCS and NOT specifying a `default_time_to_live`, then we use the already defined table TTL. For any other condition where the user is specifying the `default_time_to_live` we check against the user provided input.

The reason is quite simple: If we don't do that (ie: We always check from the pre-existing table), then we block any other table `ALTER` statement needlessly. We don't want an `ALTER TABLE x WITH comment='bla'` to fail just because the table was incorrectly created from the beginning, right? ;-)

Reply:
No, I meant the `sstable_window_size`.

When I reviewed the code above, I was worried (and wrote a comment) that you always take `sstable_window_size` from cfprops (the ALTER command), never from the existing table. This means (I think) that if the table was previously created with some non-default `sstable_window_size`, and now you get an ALTER command which modifies `default_time_to_live`, you calculates will use the default `sstable_window_size` instead of the actual one.

After suspecting this, I looked at your test and saw that you use the default `sstable_window_size` in these tests, so they won't be sensitive to such a bug if it's a real bug.

Reply:
Ok, it makes sense now. You're right, a change in `default_time_to_live` is currently checked against the default, duh. I'll prepare a fix.